### PR TITLE
update scipy dependency

### LINKED ISSUE #226: SciPy-1.6 version breaks WXRotation
The new SciPy release is not compatible with our `WXRotations` object. Some interfaces changed and some functions seem to be removed or renamed.

### PR DESCRIPTION
**Pull request by CagtayFabry**
## Changes
exclude scipy versions related to bug #226

## Related Issues
Closes #226

## Checks

- [x] updated CHANGELOG.md
- [ ] updated tests
- [ ] updated doc/
- [ ] update example/tutorial notebooks

### CODE REVIEW

**Comment by CagtayFabry** [comment]
test run fine with scipy 1.6.2 🥳 

do you know a way to specify `>=1.4,!=1.6.0,!=1.6.1` in a conda recipe @marscher ?

**Comment by marscher** [comment]
I created a PR for the feedstock. You should have a pending (automatized) review request over there.

**Comment by marscher** [comment]
This can be merged and is already available on conda-forge :)

File: doc/nitpick_ignore [hunk 1]
```diff
@@ -28,6 +28,8 @@ py:class scipy.spatial.transform.Rotation
 # derived from scipy
 py:method weldx.transformations.WXRotation.from_mrp
 py:method weldx.transformations.WXRotation.random
+py:obj weldx.transformations.WXRotation.from_mrp
```
Comment (CagtayFabry) [inline doc/nitpick_ignore:31]: what do I have to put here to get rid of the following warning @vhirtham ?
`:docstring of weldx.transformations.WXRotation.from_mrp:12: WARNING: py:obj reference target not found: mrp[i]`

Comment (vhirtham) [reply doc/nitpick_ignore:31]: usually exact the same thing that is written after the `:`

File: setup.cfg [hunk 2]
```diff
@@ -40,7 +40,7 @@ install_requires =
     numpy >=0.18
     pandas >=1.0
     xarray >=0.15
-    scipy <1.6
+    scipy >=1.4,!=1.6.0,!=1.6.1
```
Comment (vhirtham) [inline setup.cfg:43]: Maybe just `>1.6.1` in the future?

**Review by vhirtham — APPROVED** [review]

File: setup.cfg (hunk 2 above)
Comment (CagtayFabry) [reply setup.cfg:43]: certainly at some point, yes :)

File: doc/nitpick_ignore (hunk 1 above)
Comment (CagtayFabry) [reply doc/nitpick_ignore]: nice, thanks !

**Pull request merged**